Source/iOS/OIDAuthorizationService+IOS.m

```diff
@@ -18,7 +18,13 @@
 
 #import "OIDAuthorizationService+IOS.h"
 
-#import "OIDExternalUserAgentIOS.h"
+#if TARGET_OS_MACCATALYST
+  #import "OIDExternalUserAgentCatalyst.h"
+  typedef OIDExternalUserAgentCatalyst ExternalUserAgent;
```

Comment:
The `typedef` pattern doesn't look right to me.

I have another suggestion for the changes to this file, and to `OIDAuthState+IOS.m`:

1. Import *both* user agents. They are #ifdef protected anyway so there's no harm in importing them.
2. Use an `#if TARGET_OS_MACCATALYST` to determine which external agent is created.

My preference for this approach is that the Catalyst-specific tweaks are localized to just the specific area of code where they matter.

Reply:
Thanks for the suggestion! I initially did that for `OIDAuthState+IOS.m` to avoid adding `#if TARGET_OS_MACCATALYST` at multiple places.

> My preference for this approach is that the Catalyst-specific tweaks are localized to just the specific area of code where they matter.
I agree, we see better now where the catalyst platform specific code is.